operatoringress/v1/types.go

```diff
@@ -56,6 +56,10 @@ type DNSRecordSpec struct {
 type DNSRecordStatus struct {
 	// zones are the status of the record in each zone.
 	Zones []DNSZoneStatus `json:"zones,omitempty"`
+
+	// observedGeneration is the most recent generation observed.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
It seems like different zones could have different observed generations.  What happens if some zones are generation 8 and others are at generation 12?

Reply:
Maybe it would be worthy to move this to DNSZoneCondition to match the direction upstream is heading with https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1623-standardize-conditions/README.md

Reply:
> It seems like different zones could have different observed generations. What happens if some zones are generation 8 and others are at generation 12?

The current logic is that the controller will see that the generation has changed, update each zone, and update the status for each zone.  If updating a zone fails, then the status for that zone will reflect that the update failed, and the next time the controller's control loop runs, it sees that it has a zone with status condition `Failed=True` and retries the update.

> Maybe it would be worthy to move this to DNSZoneCondition to match the direction upstream is heading with https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1623-standardize-conditions/README.md

￼That's an interesting proposal.  As described above, the controller can do the right thing with a single `status.observedGenerate` field, but would you like me to change it to `status.zones[*].observedGeneration` for consistency with the KEP you cited?